src/MiniProfiler.Shared/Timing.cs

```diff
@@ -77,28 +78,76 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public List<Timing> Children
```

Comment:
As an example, if you have a large profiling tree then the serialization at the end will pay a very heavy toll in duplication...for no reason in the situation. We'd just be generating `n` copies of structures as part of a normal operation that has no racing at all.

What the lock is trying to protect is the actual reference race, but there's a premium price on the member access within that reference as a result.